### ISSUE TITLE
IList<VssWMFileDescriptor^>^ VssExamineWriterMetadata::ExcludeFiles::get() is empty

### ISSUE THREAD

**Issue by guyte** [alphaleonis/AlphaVSS#3]
Consider the following:

```C#
                    IList<IVssExamineWriterMetadata> writers = _backup.WriterMetadata;
                    foreach (IVssExamineWriterMetadata metadata in writers)
                    {
                        //metadata.ExcludeFiles is always empty

                    }
```

The bug is in:
   IList<VssWMFileDescriptor^>^ VssExamineWriterMetadata::ExcludeFiles::get()

And the fix is:
```C++
   IList<VssWMFileDescriptor^>^ VssExamineWriterMetadata::ExcludeFiles::get()
   {
      if (m_excludeFiles != nullptr)
         return m_excludeFiles;

      UINT cIncludeFiles, cExcludeFiles, cComponents;
      CheckCom(mExamineWriterMetadata->GetFileCounts(&cIncludeFiles, &cExcludeFiles, &cComponents));

      List<VssWMFileDescriptor^>^ list = gcnew List<VssWMFileDescriptor^>(cExcludeFiles); //List instead of IList
      for (UINT i = 0; i < cExcludeFiles; i++)
      {
         IVssWMFiledesc *filedesc;
         CheckCom(mExamineWriterMetadata->GetExcludeFile(i, &filedesc));
         list->Add(CreateVssWMFileDescriptor(filedesc));
      }
      m_excludeFiles = list->AsReadOnly(); // Return ReadOnly copy
      return m_excludeFiles;
   }
```

**Comment by alphaleonis** [comment]
I cannot reproduce the bug you are describing here. metadata.ExcludeFiles contains information on all attempts I've tried, and I fail to see why making the list read-only would do anything to change the behavior of this method.

Do you have a more complete example that illustrates the problem? (`AlphaShadow lwm` also lists excludefiles correctly).

**Comment by guyte** [comment]
Interesting. I have experienced the issue when running from debugger (version 1.2.4000.3). ExcludeFiles were always empty. My guess was that returning as ReadOnly + converting IList to List would result in the list being copied instead of returning referenced objects (C++ is not really my thing).
I have just checked and alphashadow.exe does return the ExcludeFiles. 
I will take the 1.3 branch and give it another test.

Thanks !
